modules/node/src/transfer/transfer.service.ts

```diff
@@ -201,16 +202,47 @@ export class TransferService {
       throw new Error(`App with provided hash has not been installed: ${linkedHash}`);
     }
 
+    const freeBalanceAddr = this.cfCoreService.cfCore.freeBalanceAddress;
+
     const freeBal = await this.cfCoreService.getFreeBalance(
       userPubId,
       channel.multisigAddress,
       assetId,
     );
+    if (freeBal[freeBalanceAddr].lt(amountBN)) {
+      // request collateral and wait for deposit to come through
+      // TODO: expose remove listener
+      await new Promise(async (resolve, reject) => {
+        this.cfCoreService.cfCore.on(
+          NODE_EVENTS.DEPOSIT_CONFIRMED,
+          (msg: DepositConfirmationMessage) => {
+            if (msg.from !== this.cfCoreService.cfCore.publicIdentifier) {
+              return;
+            }
+            if (msg.data.multisigAddress !== channel.multisigAddress) {
+              return;
+            }
+            resolve();
+          },
+        );
+        try {
+          const result = await this.channelService.requestCollateral(userPubId, assetId, amountBN);
+          if (!result) {
```

Comment:
Maybe we should check something like `if (typeof result !== Promise && !result) ...` just for safety?